src/Kros.AspNetCore/Authorization/ServiceCollectionExtensions.cs

```diff
@@ -10,121 +10,124 @@
-            {
-                builder = services.AddAuthentication(schemeList.First().SchemeName);
-            }
-            else
-            {
-                builder = services.AddAuthentication();
-            }
+        if (schemeList.Count() == 1)
+        {
+            builder = services.AddAuthentication(schemeList.First().SchemeName);
+        }
+        else
+        {
+            builder = services.AddAuthentication();
+        }
 
-            foreach (var scheme in schemeList)
+        foreach (var scheme in schemeList)
+        {
+            builder = builder.AddJwtBearer(scheme.SchemeName, x =>
             {
-                builder = builder.AddJwtBearer(scheme.SchemeName, x =>
+                x.RequireHttpsMetadata = scheme.RequireHttpsMetadata;
+                x.SaveToken = false;
+                x.TokenValidationParameters = new TokenValidationParameters
                 {
-                    x.RequireHttpsMetadata = scheme.RequireHttpsMetadata;
-                    x.SaveToken = false;
-                    x.TokenValidationParameters = new TokenValidationParameters
-                    {
-                        ValidateIssuerSigningKey = true,
-                        IssuerSigningKey = new SymmetricSecurityKey(Encoding.ASCII.GetBytes(scheme.JwtSecret)),
-                        ValidateIssuer = false,
-                        ValidateAudience = false
-                    };
+                    ValidateIssuerSigningKey = true,
+                    IssuerSigningKey = new SymmetricSecurityKey(Encoding.ASCII.GetBytes(scheme.JwtSecret)),
+                    ValidateIssuer = false,
+                    ValidateAudience = false
+                };
 
-                    configureOptions?.Invoke(x);
-                });
-            }
-
-            return builder;
+                configureOptions?.Invoke(x);
+            });
         }
 
-        /// <summary>
-        /// Configure API key Basic authentication.
-        /// </summary>
-        /// <param name="builder">Authentication builder.</param>
-        /// <param name="configuration">Configuration.</param>
-        public static AuthenticationBuilder AddApiKeyBasicAuthentication(
-            this AuthenticationBuilder builder,
-            IConfiguration configuration)
+        return builder;
+    }
+
+    /// <summary>
+    /// Configure API key Basic authentication.
+    /// </summary>
+    /// <param name="builder">Authentication builder.</param>
+    /// <param name="configuration">Configuration.</param>
+    public static AuthenticationBuilder AddApiKeyBasicAuthentication(
+        this AuthenticationBuilder builder,
+        IConfiguration configuration)
+    {
+        ApiKeyBasicAuthenticationOptions configOptions = configuration.GetSection<ApiKeyBasicAuthenticationOptions>()
+            ?? throw new ArgumentNullException(nameof(configuration),
+                $"{nameof(ApiKeyBasicAuthenticationOptions)} not found in configuration");
```

Comment:
Viem, že to tak bolo už predtým, ale hláška nie je dobrá, pretože kľúč v konfigurácii nie je `ApiKeyBasicAuthenticationOptions`, ale iba `ApiKeyBasicAuthentication`. Takže správne to má byť:

```suggestion
                $"{Helpers.GetSectionName<ApiKeyBasicAuthenticationOptions>()} not found in configuration");
```